pkg/stringset/set.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package stringset
+
+import (
+	cuckoo "github.com/seiflotfy/cuckoofilter"
+)
+
+type Set interface {
+	Has(string) bool
+	HasAny([]string) bool
+}
+
+type fixedSet struct {
+	cuckoo *cuckoo.Filter
+}
+
+func (f fixedSet) HasAny(strings []string) bool {
+	for _, s := range strings {
+		if f.Has(s) {
+			return true
+		}
+	}
+	return false
+}
+
+func NewFromStrings(items ...string) Set {
+	f := cuckoo.NewFilter(uint(len(items)))
+	for _, label := range items {
+		f.InsertUnique([]byte(label))
+	}
+
+	return &fixedSet{cuckoo: f}
+}
+
+func (f fixedSet) Has(s string) bool {
+	return f.cuckoo.Lookup([]byte(s))
+}
+
+type mutableSet struct {
+	cuckoo *cuckoo.ScalableCuckooFilter
+}
+
+type MutableSet interface {
+	Set
+	Insert(string)
+}
+
+func New() MutableSet {
```

Comment:
Maybe having some comment on when to use Set vs MutableSet would be good?

Reply:
Do we even get benefits in keeping the fixed set? Seems like we could just keep the `MutableSet` and let the consumer decides on which parts of the set they want to depend on.

So Thanos Sidecar would rely on some sort of `ReadOnlySet` interface and other on a `MutableSet` interface, both backed by the same struct.

Reply:
The fixed implementation has a better estimate on the size of the underlying filter, whereas the mutable filter grows the slice by a factor of 2 if it needs more space. 

I would also prefer not to have redundant comments such as `MutableSet is a set that can be modified`.